Draft of Removing Restriction to Access to Multiplier and Allowing Unrestricted Minting of Datacap (#803)
filecoin-project/FIPs: FIPS/fip-0078.md

The more transparent, predictable and fair Filecoin that we propose will attract new miners to the network, end malinvestment into the storage of unmonetizable datasets, and free up a large amount of money and man-hours that can be spent on more productive endeavors like improving the tech and business development with paying users. 

## Specification
 **Change in the Datacap Actor's Mint Function**: Alter the DataCap actor's mint() function to remove the restriction that the caller must be the "governor" (the verified registry actor). Allow any actor to mint datacap.

>  **Snap Deals**
> When an SP only wants to commit capacity, but do so with a high multiplier, the verified allocation they claim must specify a data CID, which must of course match the committed data. (Before direct onboarding, they would do a full-on deal with the built-in market; after direct onboarding, they can just directly commit the data). This would have the effect of giving the sector verified deal weight, and making it not appear to be CC, and therefore not eligible for subsequent snap deal with client data.
> 
> We will work around this by detecting explicit commitments to the "Zero" unsealed CID, which is the same unsealed CID that CC sectors use. Even though this unsealed CID is not stored on chain, we can set a flag in sector on-chain info to indicate that the sector data is zero, despite having verified deal weight, and that it's therefore a valid target for snap deals.

Reviewer: ```suggestion
 **Snap Deals**: When an SP only wants to commit capacity, but do so with a high multiplier, the verified allocation they claim must specify a data CID, which must match the committed data. (Before direct onboarding, they would do a full-on deal with the built-in market; after direct onboarding, they can just directly commit the data). This would give the sector verified deal weight, making it not appear to be CC, and therefore ineligible for subsequent snap deal with client data. We will work around this by detecting explicit commitments to the "Zero" unsealed CID, which is the same unsealed CID that CC sectors use. Even though this unsealed CID is not stored on chain, we can set a flag in the sector's on-chain info to indicate that the sector data is zero, despite having verified deal weight, and that it is a valid target for snap deals.
```

## Design Rationale
While more comprehensive solutions might be considered in the future, this FIP offers an immediate, simpler approach. 

Allowing any actor to mint datacap is the simplest change to the current protocol that ends human gating of datacap issuance. Datacap can then be transferred to the verified registry to create a datacap allocation. Storage Providers can claim the allocation assigned to them and get the associated 10x deal multiplier when onboarding the data piece.

> After this change, the verified registry and datacap actors will be mostly obsolete. Allowing unlimited datacap does not create any complexity. Future protocol improvements could remove these actors and support direct specification of a multiplier when onboarding to further simplify the protocol and reduce cost.

Reviewer: ```suggestion

After this change, the verified registry and datacap actors will be mostly obsolete. Allowing unlimited datacap does not create any complexity. Future protocol improvements could remove these actors and support direct specification of a multiplier when onboarding, in order to further simplify the protocol and reduce cost.
```

## Backwards Compatibility

> This proposal does not change any chain state or method interfaces so is backwards compatible to all clients. All existing workflows will continue to work, and this change is also compatible with direct onboarding. A network upgrade is required to deploy the new actor code.

Reviewer: ```suggestion
This proposal does not change any chain state or method interfaces. It is, therefore, backwards compatible with all clients. All existing workflows will continue to work, and this change is also compatible with direct onboarding. A network upgrade is required to deploy the new actor code.
```

> After this upgrade, Storage Providers (SPs) can engage with any deal, whether Fil+ or not by minting Datacap. Deals will still need the `verified` flag set to true, but clients will be able to freely do this because they can mint the required datacap.

Reviewer: ```suggestion
After this upgrade, Storage Providers (SPs) can engage with any deal, whether Fil+ or not, by minting Datacap. Deals will still need the `verified` flag set to true, but clients will be able to do this freely because they can mint the required datacap.
```

## Test Cases

> Test that an account actor that is not a notary can mint datacap
> Test that a non-account actor can mint datacap

Reviewer: ```suggestion
- Test that an account actor that is not a notary can mint datacap
- Test that a non-account actor can mint datacap
```

Substitution risk is also mitigated by the fact that the expiration of CC sectors is a well-established trend. Substitution may even help to retain some hardware that would otherwise completely drop off the network.

In order to get 20% of QAP, an attacker who only has CC would need about 600 petabytes of RBP and about 40,000,000 FIL. To convert 600 PB of CC to 10x sectors via snap deals would take over 10,000 sealers working non-stop for a week. This is a worst-case analysis that assumes the attacker is the only one to substitute CC for 10x sectors; in reality, this is extremely unlikely.

The effects of this change on RBP, QAP and network security were also analyzed by CEL [in this report](https://hackmd.io/zp-40inqSny5HMqmTSOKxw?view)

## Incentive Considerations

> The proposed change removes Fil+'s incentive for the onboarding of non-random data. As a result, it's anticipated that the onboarding rate of non-random data may decrease. It is still uncertain how much of the data currently incentivized by Fil+ is fake.

Reviewer: ```suggestion
The proposed change removes Fil+'s incentive for the onboarding of non-random data. As a result, it's anticipated that the onboarding rate of non-random data may decrease. It remains unclear how much of the data currently incentivized by Fil+ is not in compliance with the program's goals.
```

_Fake_, like _useful_, remains vaguely and contentiously defined.

With the removal of this restriction, effectively-empty sectors can now benefit from a multiplier. This adjustment allows Storage Providers (SPs) to onboard random data at a profit, potentially leading to an increase in such sectors being onboarded and growth in RBP.

Following this amendment, the primary impetus for SPs to onboard data will originate from outside the protocol itself. It could include direct payments from data clients or new incentive models implemented in FVM contracts.

Because all sectors will have the same multiplier, we expect miners to charge more for storing user data. So users whose data is currently stored via Fil+ deals will probably have to pay more for storage after the sector containing their data expires. We suggest existing users discuss pricing changes with their SPs or the intermediary who handled deal-making for them. It is unlikely that user data will be dumped before sector expiry since a miner would have to pay a termination fee and re-seal a new sector. 

Data onboarded after the transition to free multipliers will not face the risk of price hikes or dumping.  

> It is hard to estimate how much user demand Filecoin will lose in the short-run, (both demand for data already stored on the network, and data that potentially could have been onboarded in the future). There are two reasons for this: first, we don't know how much of the data stored via Fil+ deals has an actual user behind it, second, because Fil+ distorted price discovery, we don't know what users are willing to pay for Filecoin storage and what miners are willing to accept.  

Reviewer: ```suggestion
It is hard to estimate how much user demand Filecoin will lose in the short run, (both demand for data already stored on the network and for data that potentially could have been onboarded in the future). There are two reasons for this: first, we don't know how much of the data stored via Fil+ deals has an actual user behind it; second, because Fil+ distorted price discovery, we don't know what users are willing to pay for Filecoin storage and what miners are willing to accept.  
```
